apps/prairielearn/python/question_phases.py

```diff
@@ -148,6 +148,16 @@ def process_element(
                 if result is not None:
                     raise Exception("Another element already returned a file")
                 result = element_value
+            else:
+                if element_value is not None and element_value is not data:
+                    # TODO: Once this has been running in production for a while,
+                    # change this to raise an exception.
+                    sys.stderr.write(
+                        f"Function {str(phase)}() in {str(element_controller)} returned a data object other than the one that was passed in.\n\n"
+                        + "There is no need to return a value, as the data object is mutable and can be modified in place.\n\n"
+                        + "For now, the return value will be used instead of the data object that was passed in.\n\n"
+                        + "In the future, returning a different object will trigger a fatal error."
+                    )
```

Comment:
Should use the warnings module instead of writing directly to stderror: https://docs.python.org/3/library/warnings.html#module-warnings

This should be an easy change.

Reply:
What's the benefit of using that?

Reply:
Generally the accepted style because client code can silence warnings / convert to exceptions as needed.

Reply:
Gotcha, given that there is no "client code" and we never want this to do anything other than log to stderr, I'm going to leave this as is 👍